### DISCUSSION TITLE
Unable to do interactive plots in Spyder

### DISCUSSION THREAD

**Discussion by kskollepara** [pyvista/pyvista#3980] [Q&A] [answered]
I tried to render some interactive plots from pyvista demos. However, Spyder does not do the job.

Please let me know if I am missing any dependencies

```
from pyvista import demos
ex = demos.glyphs(2)
ex.plot(color='lightgrey', pbr=True, metallic=0.2,
          jupyter_backend='ipygany', interactive=True)
```

In Spyder, I get the following warning  `Ignoring ``jupyter_backend`` ` , but no plot is rendered irrespective of the backend chosen.

If I include the `notebook=True` flag, then a static plot is rendered.

Here is the output for `pv.Report()` :
```
--------------------------------------------------------------------------------
  Date: Wed Feb 08 15:24:04 2023 CET

                OS : Linux
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 15.4 GiB
       Environment : IPython
       File system : ext4
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : Quadro P620/PCIe/SSE2
       GPU Version : 4.6.0 NVIDIA 470.161.03

  Python 3.9.15 (main, Nov 24 2022, 14:31:59)  [GCC 11.2.0]

           pyvista : 0.37.0
               vtk : Module not found
             numpy : 1.23.5
           imageio : 2.19.3
            scooby : 0.7.0
             pooch : v1.4.0
        matplotlib : 3.4.2
         pyvistaqt : 0.9.0
             PyQt5 : 5.15.7
           IPython : 7.31.1
        ipyvtklink : 0.2.3
             scipy : 1.9.3
              tqdm : 4.64.1
            meshio : 5.3.4
        jupyterlab : 3.5.2
         pythreejs : Version unknown

  Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904
  for Intel(R) 64 architecture applications
--------------------------------------------------------------------------------

```

**Comment by banesullivan** [comment, marked as answer]
Unfortunately, this isn't feasible in Spyder and you need to pass `notebook=False` when instantiating a `pyvista.Plotter()` (e.g., `pyvista.Plotter(notebook=False)`) which will give you an interactive pop-up GUI. 

**_Spyder is not equivalent to a Jupyter environment._**

Spyder's IPython is not capable of displaying any HTML output and thus, you cannot use any `ipywidgets` or anything relying on HTML or JavaScript -- something all of our notebook backends requires.

If you're looking to use PyVista is something closer to an IDE, try VSCode and its interactive view outputs:

<img width="1710" alt="Screen Shot 2023-02-07 at 8 52 13 PM" src="https://user-images.githubusercontent.com/22067021/217570377-f8b530d3-40ca-4be7-b8bf-67e33f283b3a.png">
